storm-data-contracts/src/main/java/com/forter/contracts/BaseContractsBoltExecutor.java

```diff
@@ -80,7 +86,20 @@ public void execute(Tuple inputTuple, BasicOutputCollector collector) {
             }
             else {
                 TInput input = (TInput) validatedInputContract.getContract();
-                output = delegate.execute(input);
+                boolean isCachSupported = isInputCached && cache.isPresent();
+                Optional<TOutput> cachedOutput = Optional.absent();
+                if (isCachSupported) {
+                    cachedOutput = cache.get().get(input);
+                }
+                if (cachedOutput.isPresent()) {
+                    output = cachedOutput.get();
+                } else {
+                    long startTime = System.currentTimeMillis();
```

Comment:
are you sure that startTime should be measured before calling execute, and not after?